src/lib/registration/yaml_product.rb

```diff
@@ -0,0 +1,35 @@
+# allow to register against products defined in yaml. Very similar concept to online medium just this time
+# for first boot purpose
+
+require "yaml"
+
+module Registration
+  class YamlProduct
```

Comment:
The class name is good enough for now, later it will have to be improved. "YamlProduct" reflects the source, not the purpose.

I've collected the existing related classes:
- [Y2Packager::Product](https://www.rubydoc.info/github/yast/yast-yast2/master/Y2Packager/Product)
- [Y2Packager::ProductSpec](https://www.rubydoc.info/github/yast/yast-packager/master/Y2Packager/ProductSpec)
- [Y2Packager::ControlProductSpec](https://www.rubydoc.info/github/yast/yast-packager/master/Y2Packager/ControlProductSpec)
- [Y2Packager::LibzyppProductSpec](https://www.rubydoc.info/github/yast/yast-packager/master/Y2Packager/LibzyppProductSpec)
- [Y2Packager::RepoProductSpec](https://www.rubydoc.info/github/yast/yast-packager/master/Y2Packager/RepoProductSpec)

Reply:
well, we are very close to that control spec as we are defined also in some file, but I worry where we collide if we start using the existing one